### ISSUE TITLE
Wrong package name in the ESS module

### ISSUE THREAD

**Issue by robbert-vdh** [hlissner/doom-emacs#880]
The package name used in `lang/ess/config.el` does not match the name of the actual package declared in `packages.el` (`ess-mode` should just be `ess`). Also, the `ess-doc-map` key definitions have to be deferred as `ess-doc` doesn't exist until `ess-help` has been loaded.

### Observed behavior

Nothing from the `:config` part of the use-package declaration gets loaded.

### Expected behavior

The configuration should get applied.

### Steps to reproduce

1. Run the `R` command or open a `.R` file.

### System information

<details>
<summary>Click to expand</summary>

- OS: gnu/linux (x86_64-pc-linux-gnu)
- Emacs: 26.1 (jul 05, 2018)
- Doom: 2.0.9 (develop 75036cefb02503a2c93cdfd6d051a5bc0519065c)
- Graphic display: t (daemon: t)
- System features: XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND GPM DBUS GSETTINGS NOTIFY ACL GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 MODULES THREADS LIBSYSTEMD LCMS2
- Details:
  ```elisp
  elc count: 28
  uname -a:  Linux desktop 4.18.6-arch1-1-ARCH #1 SMP PREEMPT Wed Sep 5 11:54:09 UTC 2018 x86_64 GNU/Linux
  modules:   (:feature eval (evil +everywhere) file-templates (lookup +devdocs +docsets) snippets spellcheck (syntax-checker +childframe) workspaces :completion (helm +fuzzy) :ui doom-modeline doom-quit evil-goggles hl-todo nav-flash treemacs (popup +all +defaults) vc-gutter vi-tilde-fringe window-select :editor multiple-cursors rotate-text :emacs ediff electric eshell hideshow imenu term vc :tools ein docker gist make magit pdf prodigy rgb upload :lang data emacs-lisp ess (haskell +intero) javascript latex (markdown +pandoc) (org +attach +babel +capture +export +present) php python rest rust (sh +fish) web :app :config)
  packages:  (atomic-chrome company-lsp (company-racer :ignore t :disable t) ein evil-lion ggtags kotlin-mode (lsp-css :recipe (lsp-css :fetcher github :repo emacs-lsp/lsp-css)) lsp-intellij lsp-mode lsp-rust lsp-ui page-break-lines phpcbf pkgbuild-mode (racer :ignore t :disable t))
  exec-path: (~/.cargo/bin ~/.local/bin /usr/local/sbin /usr/local/bin /usr/bin /usr/lib/emacs/26.1/x86_64-pc-linux-gnu)
  ```
</details>

**Comment by hlissner** [comment]
Thanks for letting me know! It appears ESS has changed upstream. As of ae6d106 this should be fixed. Let me know if that isn't the case.

**Comment by robbert-vdh** [comment]
I've been messing around with ESS a bit and I noticed a few other inconsistencies such as incorrect company backends and ESS buffers being cleaned up prematurely as they're not considered 'real' buffers. Shall I make a separate issue about these things? I've fixed some of them in my [personal config](https://github.com/robbert-vdh/dotfiles/blob/faca198fb139b2bc883abcc2ec73f4301920f7ba/user/emacs/.config/doom/config.el#L123-L135).
